SQLQueryGroupProjectTeamB1Julia.sql

```diff
@@ -0,0 +1,255 @@
+
+
+INSERT INTO tblQUARTER(QuartName, QuartBegin, QuartEnd, QuartDescr)
+VALUES ('Winter', '01-06-2020', '03-13-2020', 'Winter quarter is the quarter in the beginning of the new year - it is usually the second quarter of the new school year - the weather can be rainy or snowy'),
+		 ('Spring', '03-30-2020', '06-05-2020', 'Spring quarter is usually the third quarter of the new school year - the weather is typically rainy and sunnier towards the end'),
+		 ('Summer', '06-22-2020', '08-21-2020', 'Summer quarter is usually a quarter off for students - it has the warmest weather out of all the quarters'),
+		 ('Summer A', '06-22-2020', '07-22-2020', 'Summer A is the quarter during the first half of summer'),
+		 ('Summer B', '07-23-2020', '08-21-2020', 'Summer B is the quarter during the second half of summer'),
+		 ('Fall', '09-30-2020', '12-11-2020', 'Fall quarter is the quarter in the beginning of the new school year - it is usually the first quarter of the new school year - the weather can be warm, rainy, snowy')
+
+INSERT INTO tblHOUR(HourName)
+VALUES(0), (1), (2), (3), (4), (5), (6), (7), (8), (9), (10), (11), (12), (13), (14), (15), (16), (17), (18), (19), (20), (21), (22), (23)
+GO
+
+--BUSINESS RULES
+
+--Enforce the business rule preventing employees to be scheduled on their birthday
+
+CREATE FUNCTION dbo.BR_NoShiftsOnEmpBirthday()
+RETURNS INT
+AS
+BEGIN
+	DECLARE @RET INT = 0
+	IF EXISTS(SELECT S.ShiftID
+			  FROM tblSHIFT S
+				JOIN tblEMP_SHIFT_STATUS ESS
+					ON S.ShiftID = ESS.ShiftID
+				JOIN tblEMPLOYEE_POSITION EP
+					ON ESS.EmpPosID = EP.EmpPosID
+				JOIN tblEMPLOYEE E
+					ON EP.EmployeeID = E.EmployeeID
+				JOIN tblDAY D
+					ON D.DayID = S.DayID
+				JOIN tblMONTH M
+					ON M.MonthID = S.MonthID
+			  WHERE DAY(E.DOB) = D.[DayName]
+				AND MONTH(E.DOB) = M.MonthNum
+			 )
+	BEGIN
+			SET @RET = 1
+	END
+	RETURN @RET
+END
+GO
+
+ALTER TABLE tblSHIFT
+ADD CONSTRAINT BR_NoShiftsOnBirthday
+CHECK(dbo.BR_NoShiftsOnEmpBirthday( ) = 0)
+GO
+
+--Enforce the business rule preventing multiple shifts of the same day/time to be scheduled - one employee per shift
+
+CREATE FUNCTION dbo.BR_NoScheduleShiftsAtSameTime()
+RETURNS INT
+AS 
+BEGIN
+	DECLARE @RET INT = 0
+	IF EXISTS(SELECT ESS.ESSID, COUNT(ESS.EmpPosID)
+			  FROM tblEMP_SHIFT_STATUS ESS
+				JOIN tblSHIFT S
+					ON ESS.ShiftID = S.ShiftID
+				JOIN tblQUARTER Q
+					ON Q.QuarterID = S.QuarterID
+				JOIN tblMONTH M
+					ON M.MonthID = S.MonthID
+				JOIN tblDAY D
+					ON D.DayID = S.DayID
+				JOIN tblHOUR H
+					ON H.HourID = S.BeginHourID
+				JOIN tblHOUR H2
+					ON H2.HourID = S.EndHourID
+				JOIN tblLOCATION L
+					ON L.LocationID = S.LocationID
+			  GROUP BY S.ShiftID, S.QuarterID, S.MonthID, S.DayID, S.BeginHourID, S.EndHourID, S.LocationID
+			  HAVING COUNT(ESS.EmpPosID) > 1
+			)
+	BEGIN
+		SET @RET = 1
+	END
+	RETURN @RET
+END
+GO
+
+ALTER TABLE tblSHIFT
+ADD CONSTRAINT BR_NoShiftsSameTime
+CHECK(dbo.BR_NoScheduleShiftsAtSameTime( ) = 0)
+GO

```

Comment:
The second business rules do not hold when there is a ‘swapped’ shift. Since if it is swapped, it will always return 1. 

In addition, if the business rule is to restrict employees being assigned to a shift, should it be tblEMP_SHIFT_STATUS to be altered?